src/legacy/core_plugins/kibana/public/dashboard/migrations/migrations_730.ts

```diff
@@ -38,8 +39,36 @@ export function migrations730(
     doc.attributes.kibanaSavedObjectMeta.searchSourceJSON = JSON.stringify(
       moveFiltersToQuery(searchSource)
     );
+  } catch (e) {
+    // eslint-disable-next-line
+    console.error(
+      'Migration error encountered while trying to migrate queries out of filters array: ',
+      e
+    );
     return doc;
```

Comment:
I'm not sure you want to return the doc in this case. Returning it will make Kibana think that it is properly migrated. Is that OK?

Reply:
I don't like this pattern at all, but it seems like all/most of the current migrations followed this pattern.   I created an issue for it - https://github.com/elastic/kibana/issues/38669

I decided not to switch to that for this PR though as I thought it should be focused and done for everything... and also I'm a little nervous of it. I think that'll mean data may break in a minor that previously worked?

Reply:
you could use the migrations logger in this case:

https://github.com/elastic/kibana/blob/master/src/legacy/core_plugins/kibana/migrations/migrations.js#L191